File: admin/src/main/java/com/pedro/fitchaadmin/machine/dtos/CreateMachineDTO.java
package com.pedro.fitchaadmin.machine.dtos;

import java.util.List;

public record CreateMachineDTO(
        String slug,
        String name,
        String description,
        String photo,
        String categoryKey,
        String substitutionGroup,
        String trackingType,
        boolean requiresWeight,
        List<String> aliases
) {
}

File: backend/internal/dtos/machine/createMachine.dto.go
package dtos

type CreateMachineType struct {
	Name           string `json:"name"`
	Description    string `json:"description"`
	Photo          string `json:"photo"`
	CategoryKey    string `json:"categoryKey"`
	TrackingType   string `json:"trackingType"`
	RequiresWeight *bool  `json:"requiresWeight"`
}
